File: scaffolds/functions/kotlin/Handler.kt
// {name} — Flux function (compiled to WASM via Kotlin/Wasm)
// Build: ./gradlew wasmWasiJar
import kotlinx.serialization.json.*

@Suppress("unused")
@WasmExport("{name}_handler")
fun {Name}Handler(inputPtr: Int, inputLen: Int): Long {
    // TODO: decode input JSON at inputPtr
    val response = """{ "ok": true }"""
    val bytes    = response.encodeToByteArray()
    // Allocation: host is responsible for freeing.
    return (inputPtr.toLong() shl 32) or bytes.size.toLong()
}

File: scaffolds/functions/csharp/Handler.cs
// {name} — Flux function (compiled to WASM via dotnet wasi)
// Requires: dotnet add package Wasi.Sdk
using System.Text.Json;
using System.Runtime.InteropServices;

public static class {Name}Handler
{
    [UnmanagedCallersOnly(EntryPoint = "{name}_handler")]
    public static long Handle(IntPtr inputPtr, int inputLen)
    {
        // TODO: decode input, run logic
        var output = JsonSerializer.SerializeToUtf8Bytes(new { ok = true });
        var outPtr = Marshal.AllocHGlobal(output.Length);
        Marshal.Copy(output, 0, outPtr, output.Length);
        return ((long)outPtr << 32) | output.Length;
    }
}
